### ISSUE TITLE
Add @props to components

### ISSUE THREAD

**Issue by zimt28** [mhanberg/temple#63]
I think there's no way to access the list of props given to a component. It would be nice to have something like `@props` available. What do you think? I'd happily send a PR.

**Comment by mhanberg** [comment]
I think this would be a good enhancement, feel free to send a PR!

**Comment by mhanberg** [comment]
Do you mind also sharing your use case for this enhancement?

**Comment by zimt28** [comment]
Sure. It's mainly for wrapper components, like a custom link component. Currently I do this:

```elixir
# Hint: This syntax doesn't work using the official Temple repo
defcomponent header_dropdown_menu_link(props) do
  a Keyword.get(props, :title),
    [
      href: Keyword.get(props, :to),
      class: "block py-1 px-4 text-gray-800 hover:bg-gray-200"
    ] ++
      Keyword.drop(props, [:title, :to])
end
```
This way I can still add custom attributes to the link:
```elixir
header_dropdown_menu_link(
  title: "Sign Out",
  to: Routes.session_path(@conn, :delete),
  data_no_instant: nil,
  id: "sign_out"
)
```
Adding `@props` allows me to switch back to the official Temple repo.

**Comment by mhanberg** [comment]
Closed by #66
